include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp

```diff
@@ -57,12 +57,37 @@ struct make_aligned_sequence_type
     // seqan3::gap_decorator. Ranges that do not model std::random_access_range for instance cannot be augmented with
     // the gap_decorator and need to be copied instead.
 
+    //!\brief The unaligned sequence type that will be used to assign a sequence to seqan3::assign_unaligned.
+    using unaligned_sequence_type = decltype(std::declval<range_t>() | views::type_reduce | views::slice(0, 1));
```

Comment:
do you really want expose this type? Is that somewhere used?

Reply:
Not sure why a struct in `seqan3::detail` needs this, but I've made that change.

Reply:
It has nothing todo whether or not it is in detail. It is the question of whether or not the intermidate type should be exposed for usage also for the developer.